imap/http_caldav_sched.c

```diff
@@ -632,18 +632,45 @@ static int imip_send(const char *userid, struct sched_data *sched_data,
         jsevent = json_null();
         patch = jmapical_tojmap(sched_data->newical, NULL, NULL);
     }
+
+    struct jmap_caleventid eid = { 0 };
+    icalcomponent *comp = NULL;
+    icalproperty *prop = NULL;
+    struct buf buf = BUF_INITIALIZER;
+
+    /* if there's going to be a recurrence-id, it will be a layer down */
+    comp = icalcomponent_get_first_real_component(sched_data->itip);
+    if (comp) {
```

Comment:
The construct
```c
icalcomponent *comp = NULL;
comp = icalcomponent_get_first_real_component(sched_data->itip);
```
leads to dead initialization warning, e.g. in clang static analyzer (kind of useless assignment).  The more concise
```c
icalcomponent *comp = icalcomponent_get_first_real_component(sched_data->itip);
```
produces no such warning.

Reply:
That's interesting, can you paste the warning?  I don't have a recent clang handy

Reply:
https://mail.aegee.org/dpa/scan99/2022-08-16-122910-1822492-1/

Reply:
I would prefer just as @dilyanpalauzov to use single-line declaration and initialization.

Reply:
done